service_user.go

```diff
@@ -78,6 +85,24 @@ func (h *ServiceUsersHandler) Get(project, serviceName, username string) (*Servi
 	return nil, err
 }
 
+// Put modifies the given Service User in Aiven.
+func (h *ServiceUsersHandler) Put(project, service, username string, password string) (*ServiceUser, error) {
+	msr := ModifyServiceUserRequest{Operation: "reset-credentials", NewPassword: password, Authentication: "caching_sha2_password"}
```

Comment:
I'd make this an optional parameter. This is only relevant for MySQL so sending it for others is a bit weird (server currently doesn't seem to care about it for other services but we could start using the same field for something else for other services) and being able to specify the legacy native authentication is something that could be reasonable in some use cases. If nothing is provided the server defaults to keeping the existing authentication mechanism.

Reply:
Go does not natively support optional parameters so we would need to do something like a constructor for the `ModifyServiceUser` struct or maybe an `Options`?
As unclean as it might be, my preferred way would be to add a `PutMySQL` function that handles that use case so it doesn't expose the quirks of Go to an end user

Reply:
I was just thinking string pointer that can be nil and adding `,omitempty` into the JSON definition.